sdk/storage/Azure.Storage.DataMovement/src/StorageResource.cs

```diff
@@ -49,5 +51,14 @@ protected StorageResource()
         /// <returns>A <see cref="StorageResourceCheckpointDetails"/> containing the checkpoint information for this resource.</returns>
         [EditorBrowsable(EditorBrowsableState.Never)]
         protected internal abstract StorageResourceCheckpointDetails GetDestinationCheckpointDetails();
+
+        /// <summary>
+        /// Validates whether the proper protocol has been set for the resource.
+        /// Applies only to File Shares.
+        /// </summary>
+        /// <returns>Whether the protocol of the resource is NFS. Returns null if the resource is not File Share.</returns>
```

Comment:
nit: Let's make the docs here a bit more generic. "Ensure the transfer is valid. Intended to be called on destination resources" or something like that. Also, it doesn't return anything now so we can remove the returns tag.